ddtrace/internal/_encoding.pyi

```diff
@@ -15,16 +15,21 @@ class BufferItemTooLarge(Exception):
 
 class BufferedEncoder(object):
     content_type: str
+    def encode_item(self, item: Any) -> bytes: ...
+    def encode(self) -> Optional[bytes]: ...
+
+class ListBufferedEncoder(BufferedEncoder):
     def __init__(self, max_size: int, max_item_size: int) -> None: ...
     def __len__(self) -> int: ...
-    def put(self, trace: Trace) -> None: ...
-    def encode(self) -> Optional[bytes]: ...
+    def clear(self) -> None: ...
+    def put(self, item: Any) -> None: ...
+    def get(self) -> List[bytes]: ...
```

Comment:
I'd put `clear`, `put` and `get` in `BufferedEncoder` since their types are generic enough, and have `encode_item` in `ListBufferedEncoder`. Also, I'd keep the API as small as possible so only add methods if we have or think we'll have a use case for them (e.g. we don't seem to be using `clear` nor `get` in our code). WDYT?

Reply:
I removed `clear` but kept `get` because subclasses need it to access the list buffer without accessing the private field directly. I also added a test for the `get` method.